client/js/widgets/widget.js

```diff
@@ -1851,13 +1851,15 @@ export class Widget extends StateManaged {
       return;
 
     await this.bringToFront();
-    if(this.get('parent') && !this.currentParent)
-      this.currentParent = widgets.get(this.get('parent'));
+    const curAncestor = this.get('_ancestor');
+    if(curAncestor && !this.currentParent)
+      this.currentParent = widgets.get(curAncestor);
     if(this.currentParent != holder)
       await this.checkParent(true);
-
-    await this.set('owner',  null);
-    await this.set('parent', holder.get('id'));
+    if(this.currentParent != holder || this.get('owner') || holder.get('childrenPerOwner')) {
```

Comment:
This could probably be separate checks for each of the set calls, i.e.

```js
if (this.get('owner') || holder.get('childrenPerOwner'))
  await this.set('owner', null);
if (this.currentParent != holder)
  await this.set('parent', holder.get('id'));
```

Or do we need to set the parent if we have owner / childrenPerOwner set?

Reply:
I'm not 100% sure what we need to do when widget is owned or childrenPerOwner is set on the target holder. I need to do some tests with MOVE with from and to set to the same childrenPerOwner holder (with and without piles in the holder) to see what production does.